File: past/ABC050/C/main.nim
import algorithm,sequtils,strutils,math

var
  n: int = parseInt(stdin.readLine)
  al: seq[int] = stdin.readLine.split.map(parseInt)  
  x: int = 0
  bl: seq[int] = newSeq[int]()

al.sort do (x,y:int) -> int : result = cmp(x,y)

if n mod 2 == 0:
  for i in 0..<n:
    bl.add(2*(i div 2)+1)
else:
  bl.add(0)
  for i in 0..<n-1:
    bl.add(2*(i div 2)+2)
    
if al != bl:
  echo 0
else:
  var result = 1
  # var va
  for  _ in 0..<(n div 2):
    result = result * 2
    # result = result mod (1e9.int + 7)
  echo result
  # let ans = 1 shl (n div 2)
  # echo ans mod (1e9.int + 7)
  # ((1 << (n // 2)) % ((10 ** 9) + 7))
  # (1 << (n // 2)) % ((10 ** 9) + 7)

File: past/ABC050/C/main.py
n = int(input())
m = list(map(int, input().split()))

l = {}
ll = {}

if n % 2 == 0:
  for i in m:
    l[2*(i//2)+1] = 2
    if i in ll:
      ll[i] += 1
    else:
      ll[i] = 1
else:
  for i in m:
    if 2*(i//2) == 0:
      l[2*(i//2)] = 1
    else:
      l[2*(i//2)] = 2
    if i in ll:
      ll[i] += 1
    else:
      ll[i] = 1

if l != ll:
  print(0)
  exit()

print((1 << (n // 2)) % ((10 ** 9) + 7))
